src/pages/wallet/AssetChain.tsx

```diff
@@ -29,11 +30,12 @@ const AssetChain = (props: Props) => {
   const { data: prices, ...pricesState } = useExchangeRates()
   const { t } = useTranslation()
   const networkName = useNetworkName()
+  const allNetworks = useNetworks().networks[networkName]
 
   const networks = useNetwork()
   const { devMode } = useDevMode()
 
-  const { icon, name } = networks[chain] || {}
+  const { icon, name } = allNetworks[chain] ?? { name: chain }
```

Comment:
@alecande11 just want to make sure, you want to replace the chain name of unknown chains with the chain name for the particular network on Terra?

Reply:
No, with their chainID